polytracker/src/taintdag/fnmapping.cpp

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2022-present, Trail of Bits, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed in accordance with the terms specified in
+ * the LICENSE file found in the root directory of this source tree.
+ */
+
+#include "taintdag/fnmapping.h"
+
+#include <algorithm>
+#include <cstring>
+#include <string_view>
+
+namespace taintdag {
+
+namespace {
+
+using offset_t = FnMapping::offset_t;
+using length_t = FnMapping::length_t;
+using header_t = FnMapping::header_t;
+using index_t = FnMapping::index_t;
+
+} // namespace
+
+FnMapping::FnMapping(char *begin, char *end)
+    : map_begin(begin), map_end(end), headers_end(map_begin),
+      names_begin(map_end) {}
+
+std::optional<offset_t> FnMapping::write_name(std::string_view name) {
```

Comment:
Now that we have multiple sections in the output file that are strings (or variable length entries), maybe we should introduce a generic string section instead (to not have to replicate the same code)? If so, probably a follow up PR.

Reply:
Agreed. We should also introduce a common `Section` base class, since there's a lot of code overlap.